File: Grasshopper_GetBoxOrientation.cs
// Grasshopper C# Script: get orientation from BoxObject (DCT)
// Input: BoxObj. Outputs: P, C, X, Y, Z. Edit the path below if your Libraries folder is elsewhere.
System.Reflection.Assembly.LoadFrom("/Users/r2d2/Library/Application Support/McNeel/Rhinoceros/8.0/Plug-ins/Grasshopper (b45a29b1-4343-4035-989e-044e8580d9cf)/Libraries/DigitalCircularityToolkit.gha");

var dct = boxObj as DigitalCircularityToolkit.Objects.BoxObject;
if (dct == null) {
  Print("Input is not a BoxObject.");
  return;
}
P = dct.Plane;           // orientation plane (center + X/Y/Z axes)
C = dct.Plane.Origin;    // center
X = dct.Plane.XAxis;     // length direction (PCA1)
Y = dct.Plane.YAxis;     // width direction (PCA2)
Z = dct.Plane.ZAxis;     // height direction (PCA3)

File: Grasshopper_GetBoxOrientation.py
# Grasshopper Python: get orientation from BoxObject (DCT)
# 1. Add a "Python" or "Script" component
# 2. Input: BoxObj (item from BoxObject DCT component)
# 3. Outputs: P, C, X, Y, Z (add 5 output params)
# 4. Wire BoxObject (DCT) BoxObj output → BoxObj input

import clr
clr.AddReference("DigitalCircularityToolkit")
from DigitalCircularityToolkit.Objects import BoxObject as DCTBoxObject

box_obj = BoxObj
if not isinstance(box_obj, DCTBoxObject):
    raise ValueError("Input is not a BoxObject.")

plane = box_obj.Plane

P = plane
C = plane.Origin
X = plane.XAxis   # length (PCA1)
Y = plane.YAxis   # width (PCA2)
Z = plane.ZAxis   # height (PCA3)
